base-files: fix uid/gid auto-enumeration to avoid 16-bit limit

tew42 opened the pull request:
uid/gid range auto-set in functions.sh should start within 16bit unsigned integer range to avoid "wraparound" issues with permissions where jffs2  is employed for storage and chown 65536 (i.e., currently first auto-created user) becomes equivalent to chown 0. See #13927 for details.

The patch was tested on Archer A7 v5 running OpenWRT 23.05

(Incidentally, this should also fix unrelated but symptomatically similar permissions issues for LXC instances)

tew42 committed:
base-files: fix uid/gid auto-enumeration to avoid 16-bit limit

uid/gid range should be limited to 16bit unsigned integer range to
avoid "wraparound" issues with permissions where jffs2
is employed for storage and chown 65536 (first auto-created user)
becomes equivalent to chown 0

Fixes: #13927

Signed-off-by: Thomas Winkler <tewinkler86@gmail.com>

Ansuel commented:
@tew42 thanks a lot for the patch. 

Can you add `Fixes: #13927` right up your signed-off by tag also... also we need real email instead of noreplay github

tew42 commented:
Sigh... So I guess it's not enough to have real email in the commit but I also need to fully expose it in my account? Alright, hopefully third try is the charm

Ansuel commented:
@tew42 ehhh no idea what is the requirement for github web (i guess you are using that to propose this pr?)

tew42 commented:
@hauke thanks, that's what I did already with 26ddeb6fca0f808f382fcef2611e67ec6e84aa3c. But then the formality check kept failing (saying that it expected my anonymous no-reply email there, which I assumed it pulled from my account settings) so I also changed my github account settings/email and rebased onto main in 743d8585ffa0658faec9bf66bf70df97efb985ae to trigger another formality check. I don't get how the formality check still keeps failing with 
```"Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: Thomas Winkler <2049715+tew42@users.noreply.github.com>')
```
when I don't have that email in either the commits or my account at this point

hauke commented:
The script does not are about your github settings it only looks at the commit itself.
You can see the full commit here: https://github.com/openwrt/openwrt/pull/13937.patch It still uses the github generated mail addresses as author mail address. 

You can also use a full VS code in your browser using github code space instead of WSL: https://github.com/codespaces/new?hide_repo_select=true&ref=main-uid&repo=717229286

tew42 commented:
Many thanks for the help, now this should finally be consistent

tew42 commented:
Since @Ansuel and @hauke you were previously kind enough to take a quick look perhaps you can also get the official review going on this? Apart from my struggles with the commit message it's really quite straightforward

hauke commented:
uid and gid are only 16 bit in jffs2, see https://elixir.bootlin.com/linux/v6.6.30/source/include/uapi/linux/jffs2.h#L144
